contracts/RelayHub.sol

```diff
@@ -200,7 +200,7 @@ contract RelayHub is IRelayHub {
         // This transaction must have enough gas to forward the call to the recipient with the requested amount, and not
         // run out of gas later in this function.
         require(
-            gasleft() >= GAS_RESERVE + requiredGas,
+            initialGas >= GAS_RESERVE + requiredGas,
```

Comment:
Yes, look more right. but how much difference did it make?

Reply:
Once I fixed the server gas calculation, this is unpassable. Think about it, it required the whole tx gasLimit + GAS_RESERVE to be lower than the remaining gasLimit after some work. Impossible if the server computes the gasLimit the same way it's computed on-chain.

Reply:
This is changed in the light of relay server not being able to estimate how much gas to give to a transaction. Watch the hands: in order to pass the ```gasleft() >= GAS_RESERVE + requiredGas``` check, assuming ```gasleft := txGasLimit - gas_used_to_this_point```, the server must set `txGasLimit := GAS_RESERVE + requiredGas + gas_used_to_this_point`.

`gas_used_to_this_point` is unknown. So it is not a useful require the old way.

Reply:
Yes, and `requiredGas` is supposed to already contain `gas_used_to_this_point`